import_craft.py

```diff
@@ -0,0 +1,291 @@
+
+
+def duplicate_object_hierarchy(obj):
+    unselect_all_objects()
+    select_children(obj)
+    bpy.ops.object.duplicate(linked=True)
+    return bpy.context.scene.objects.active
+
+
+def remove_object_list(array):
+    unselect_all_objects()
+    for ob in array:
+        select_children(ob)
+    bpy.ops.object.delete()
+
+
+def read_vector(value):
+    # Unity is Y-Up, blender is Z-up
+    values_str = value.split(',')
+    return (float(values_str[0]), float(values_str[2]), float(values_str[1]))
+
+
+def read_quaternion(value):
+    # Unity is xyzw, blender is wxyz. However, Unity is left-handed and
+    # blender is right handed. To convert between LH and RH (either
+    # direction), just swap y and z and reverse the rotation direction.
+    quat_str = value.split(',')
+    return (float(quat_str[3]), -float(quat_str[0]), -float(quat_str[2]), -float(quat_str[1]))
+
+
+class CraftReader(object):
+    def __init__(self):
+        self.ship_name = ""
+        # Set of required models
+        self.prefabs = []
+        self.nb_total_parts = 0
+
+    def report(self, type_message, message):
+        print("[{}] {}".format(','.join(type_message), message))
+
+    def rename_data_elements(self, ob, directory):
+        ''' Rename the datas to avoid overriding while loading parts '''
+        unselect_all_objects()
+        rename_materials(ob)
+        rename_textures(ob)
+        rename_images(ob, directory)
+
+    def smooth_object_meshes(self, ob):
+        select_children(ob)
+        children = bpy.context.selected_objects
+        unselect_all_objects()
+        for child in children:
+            if child.type == "MESH":
+                bpy.context.scene.objects.active = child
+                bpy.ops.object.mode_set(mode='EDIT')
+                bpy.ops.mesh.faces_shade_smooth()
+                # Need to come back to OBJECT mode to avoid context error
+                bpy.ops.object.mode_set(mode='OBJECT')
+
+    def read_parts_models(self, partsdict, colliders):
+        for part in partsdict:
+            unselect_all_objects()
+            result = import_mu.import_mu(self, bpy.context, partsdict[part]['mu'], colliders)
+            if not result == {'FINISHED'}:
+                print('Warning : Error while importing file {}'.format(partsdict[part]['mu']))
+                return
+            partsdict[part]['object'] = bpy.context.scene.objects.active
+
+            self.smooth_object_meshes(partsdict[part]['object'])
+
+            directory = os.path.dirname(os.path.realpath(partsdict[part]['mu']))
+            self.rename_data_elements(partsdict[part]['object'], directory)
+
+    def apply_craft_transformations(self, part_node, blender_object):
+        ''' Apply .craft file transformations '''
+        blender_object.location = part_node['pos']
+        blender_object.rotation_mode = 'QUATERNION'
+        blender_object.rotation_quaternion = part_node['rot']
+
+    def generate_parts(self, parts, main, partsdict):
```

Comment:
really confusing to use `parts` and `partsdict`.
Also `main` usually has the semantic of "main function" as in C so `root` or something alike might be better